osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/VerifyImportedScoresCommand.cs

```diff
@@ -0,0 +1,180 @@
+                            {
+                                pp = importedScore.HighScore.pp,
+                                id = importedScore.id,
+                            });
+                        }
+
+                        elasticItems.Add(new ElasticQueuePusher.ElasticScoreItem { ScoreId = (long?)importedScore.id });
+                        Interlocked.Increment(ref fail);
+                    }
+
+                    // TODO: check data. will be slow unless we cache beatmap attribs lookups
+                    // Ruleset ruleset = LegacyRulesetHelper.GetRulesetFromLegacyId(importedScore.ruleset_id);
+                    // var referenceScore = await BatchInserter.CreateReferenceScore(ruleset, importedScore.HighScore, conn, null);
+                    // if (!check(importedScore.id, "total score", importedScore.total_score, referenceScore.TotalScore)) Interlocked.Increment(ref fail);
+                    // if (!check(importedScore.id, "legacy total score", importedScore.legacy_total_score, referenceScore.LegacyTotalScore)) Interlocked.Increment(ref fail);
+
+                    // await conn.ExecuteAsync("DELETE FROM score_performance WHERE score_id = @id; DELETE FROM scores WHERE id = @id", score, transaction);
+                    // await conn.ExecuteAsync("DELETE FROM score_legacy_id_map WHERE ruleset_id = @ruleset_id AND old_score_id = @legacy_score_id", new
+                    // {
+                    //     score.ruleset_id,
+                    //     legacy_score_id = score.ScoreInfo.LegacyScoreId
+                    // }, transaction);
+                    //
+                    // elasticItems.Add(new ElasticQueuePusher.ElasticScoreItem { ScoreId = (long?)score.id });
+                    // Interlocked.Increment(ref deleted);
+                }
+
+                if (elasticItems.Count > 0)
+                {
+                    elasticQueueProcessor.PushToQueue(elasticItems);
+                    Console.WriteLine($"Queued {elasticItems.Count} items for indexing");
+                }
+
+                lastId = importedScores.Max(s => s.id) + 1;
+
+                Console.SetCursorPosition(0, Console.GetCursorPosition().Top);
+                Console.Write($"Processed up to {lastId} ({deleted} deleted {fail} failed)");
+            }
+
+            return 0;
+        }
+
+        private bool check<T>(ulong scoreId, string name, T imported, T original)
+        {
+            if (imported == null && original == null)
+                return true;
+
+            if (imported?.Equals(original) != true)
+            {
+                Console.WriteLine($"{scoreId}: {name} doesn't match ({imported} vs {original})");
+                return false;
+            }
+
+            return true;
+        }
+
+        [SuppressMessage("ReSharper", "InconsistentNaming")]
+        private class ComparableScore
```

Comment:
The build warnings about "field never assigned to" are a bit annoying here. Easiest way to fix would probably be

```suggestion
        public class ComparableScore
```